### DISCUSSION TITLE
filter and sort question

### DISCUSSION THREAD

**Discussion by cduytran** [pocketbase/pocketbase#1866] [Q&A] [answered]
```
const eods = await pb.collection('eod').getFullList(2,{
    sort: '-created',
    filter:`eodPackRef.packNumber="${packId}"`,
    expand:'eodPackRef.packGameRef,eodTicketRef',
  })
```
I wanted to get the latest two items from collection `eod` that meets the filter requirements. The above codes return the entire collection and not even sorted correctly. What am I doing wrong?

Thanks.

**Comment by arslanastral** [comment]
You are using `getFullList` which will batch fetch all records. Your query means _get all records and make as many fetch requests needed to get two records per page_. You can see this in action in your browser console. What you want to use is:

```TS
const eods = await pb.collection('eod').getList(1,2, {
    sort: '-created',
    ...
  })
  ```
  This will return a paginated records list. In this case the first page and two items per page. Which you can access using `eods.items`.

**Reply by cduytran** [reply]
oh! Thanks. Let me try that.

**Comment by uteminhduc** [comment, marked as answer]
You must use `getList` instead of `getFullList`.
In your code, if you have 10 record with filter `getFullList` will call 5 times

_update:_
```js
const eods = await pb.collection('eod').getList(1, 2, {
    sort: '-created',
    filter:`eodPackRef.packNumber="${packId}"`,
    expand:'eodPackRef.packGameRef,eodTicketRef',
  })
```

**Reply by cduytran** [reply]
It works....!!!! Thank you.
